spectractor/simulation/simulator.py

```diff
@@ -412,7 +416,9 @@ def simulate(self, A1=1.0, A2=0., ozone=300, pwv=5, aerosols=0.05, D=parameters.
         ima2 = np.zeros_like(ima1)
         if A2 > 0.:
             nlbda_order2 = dispersion_law_order2.size
-            ima2 = self.build_spectrogram(profile_params[-nlbda_order2:], spectrum[:nlbda_order2],
+            spectrum_order1 = interp1d(lambdas,spectrum,bounds_error=False, fill_value=(0, 0))
+            spectrum_order2 = spectrum_order1(lambdas_order2) * self.lambdas_binwidths_order2 / self.lambdas_binwidths[-nlbda_order2:]
```

Comment:
maybe it is better to use simulate_spectrum to avoid arrays dimension problems ? also is there a ratio of lambda_1/lambda_2 here ?
if you switch to simulate_spectrum, as the slowest part is the atmosphere simulation, I would be in favor to be the atmopshere simulation in a separate method and interp it.

Reply:
there is no problem with dimension now, nlbda_order2 is the number of pixels with an order 2 because I remove the line "dispersion_law_order2 = dispersion_law_order2[::2]" . Even if we use simulate_spectrum we need to have a factor * self.lambdas_binwidths_order2 / self.lambdas_binwidths[-nlbda_order2:], and the calculation of the spectrum give us just an interpolation of spectrum_order1 like here (spectrum_order1(lambdas_order2)).
There is no ratio of lambda_1 / lambda_2 because here we are in ADU unit, we consider binwidths because ADU is dependent on the bin size but not on the wavelength. However, when the spectrum is extracted, there is an error on conversion ADU to FLAM because all ADU are considered with lambda = lambda_1 but it is normal and we can't change that.